frontend/javascripts/dashboard/dataset/dataset_import_view.js

```diff
@@ -97,28 +140,64 @@ class DatasetImportView extends React.PureComponent<Props, State> {
   async fetchData(): Promise<void> {
     try {
       this.setState({ isLoading: true });
-      const dataset = await getDataset(this.props.datasetId);
+      let dataset = await getDataset(this.props.datasetId);
       let dataSource;
-      let previousDataSource;
-      let dataSourceMessages = [];
       if (dataset.isForeign) {
         dataSource = await readDatasetDatasource(dataset);
+        this.setState({ savedDataSourceOnServer: dataSource });
       } else {
-        ({
-          dataSource,
+        const dataSourceSettingsStatus = {
+          appliedSuggestions: AppliedSuggestionsEnum.NoAvailableSuggestions,
+          isJSONFormatValid: IsJSONFormatValidEnum.No,
+        };
+        const {
+          dataSource: inferredDataSource,
           messages: dataSourceMessages,
-          previousDataSource,
-        } = await getDatasetDatasource(dataset));
+          previousDataSource: savedDataSourceOnServer,
+        } = await getDatasetDatasource(dataset);
+        const didParsingTheSavedDataSourceJSONSucceed =
+          savedDataSourceOnServer != null && savedDataSourceOnServer.status == null;
+        if (didParsingTheSavedDataSourceJSONSucceed) {
+          dataSource = savedDataSourceOnServer;
+          if (isDatasourceJSONValid(savedDataSourceOnServer)) {
+            dataSourceSettingsStatus.isJSONFormatValid = IsJSONFormatValidEnum.Yes;
+          }
 
-        this.setState({ previousDataSource });
-        if (previousDataSource != null && dataSource != null) {
-          const diff = diffObjects(dataSource, previousDataSource);
-          this.setState({ differenceBetweenDatasources: diff });
+          const diff = diffObjects(inferredDataSource, savedDataSourceOnServer);
+          const areObjectsEqual = _.size(diff) === 0;
+          if (!areObjectsEqual) {
+            dataSourceSettingsStatus.appliedSuggestions = AppliedSuggestionsEnum.No;
+            this.setState({ differenceBetweenDataSources: diff });
+          }
+        } else {
+          // If the current datasource json is invalid, the inferred version should be used automatically.
+          dataSource = inferredDataSource;
+          dataSourceSettingsStatus.isJSONFormatValid = IsJSONFormatValidEnum.BrokenJson;
+          dataSourceSettingsStatus.appliedSuggestions = AppliedSuggestionsEnum.Yes;
         }
+        const inferredDataSourceWithCorrectedScale = ensureValidScaleOnInferredDataSource(
+          savedDataSourceOnServer,
+          inferredDataSource,
+        );
```

Comment:
I can confirm this for the bounding box and the largest segment id, but not for the voxel size (, if this is the same as `elementClass`). This should be set in the WKW header, so it should always be correct. Therefore, you can exclude this from the ensuring method I think.

Reply:
An internal conversation showed the following results:
The backend cannot guess the bounding boxes for knossos data layers but this is fine as knossos cubes are not first class citizen.
The guessing of bounding boxes works fine for wkw datalayers.
Thus the "ensuring"-method only needs to be adjusted to ensure the usage of the saved largestSegmentId in the datasource-properties.json.

You can check whether the `largestSegmentId` is used correctly by removing the `elementClass` from the segmentation layer of the dataset you are using to experiment with.
Then the suggested settings should include the saved `largestSegmentId` instead of `0`.

Reply:
The voxel size is no in the  ensuring method :+1: